src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -658,6 +658,31 @@ public ProjectState UpdateAdditionalDocument(TextDocumentState newDocument, bool
                 latestDocumentTopLevelChangeVersion: dependentSemanticVersion);
         }
 
+        public ProjectState UpdateDocumentsOrder(ImmutableArray<DocumentId> documentIds)
+        {
+            if (documentIds.Length != _documentIds.Count)
+            {
+                throw new ArgumentException("The count of the specified document ids do not match the count of the document ids in the project.", nameof(documentIds));
+            }
+
+            var orderedDocumentIds = ImmutableList.CreateBuilder<DocumentId>();
+
+            foreach(var documentId in documentIds)
+            {
+                if (this.ContainsDocument(documentId))
+                {
+                    orderedDocumentIds.Add(documentId);
+                }
+                else
+                {
+                    throw new InvalidOperationException("A document id from the specified document ids doesn't exist within the project.");
```

Comment:
1. i believe our exception messages need to be localized.
2. generally, abbreviations are discouraged in message.
3. can we include the documentId name in the error message?

Reply:
I guess new messages need to be localized as I see some that are not.

Reply:
I'm ok with 'when in rome'.  If the existing exceptions in this file are not localized, you can stick with that.

Reply:
Too late, I localized them :D